### PR TITLE
refactor: self contain user summary and move modals

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2393](https://openedx.atlassian.net/browse/PROD-2393)
Moving all the modals out of the user summary to their separate components. No visual change.

### CODE REVIEW

**Review by DawoudSheraz — CHANGES_REQUESTED** [review]
- Move all the actions in a folder called account-actions inside user
- add tests for each component. The tests are already in UserSummary. They just need to be moved appropriately.

File: src/users/account-actions/PasswordHistory.test.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { mount } from 'enzyme';
+import React from 'react';
+
+import PasswordHistory from './PasswordHistory';
+import UserMessagesProvider from '../../userMessages/UserMessagesProvider';
+import UserSummaryData from '../data/test/userSummary';
+
+const PasswordHistoryWrapper = (props) => (
+  <UserMessagesProvider>
```
Comment (DawoudSheraz) [inline src/users/account-actions/PasswordHistory.test.jsx]: we would not need message provider here. We are not using any alert component.

File: src/users/account-actions/ResetPassword.test.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { mount } from 'enzyme';
+import React from 'react';
+
+import * as api from '../data/api';
+import ResetPassword from './ResetPassword';
+import UserMessagesProvider from '../../userMessages/UserMessagesProvider';
+import UserSummaryData from '../data/test/userSummary';
+
+const ResetPasswordWrapper = (props) => (
+  <UserMessagesProvider>
```
Comment (DawoudSheraz) [inline src/users/account-actions/ResetPassword.test.jsx]: same, no need for message provider.

File: src/users/account-actions/TogglePasswordStatus.test.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import { mount } from 'enzyme';
+import React from 'react';
+
+import * as api from '../data/api';
+import TogglePasswordStatus from './TogglePasswordStatus';
+import UserMessagesProvider from '../../userMessages/UserMessagesProvider';
+import UserSummaryData from '../data/test/userSummary';
+
+const TogglePasswordStatusWrapper = (props) => (
+  <UserMessagesProvider>
```
Comment (DawoudSheraz) [inline src/users/account-actions/TogglePasswordStatus.test.jsx]: and here too

**Review by DawoudSheraz — APPROVED** [review]
Address the nits in tests before merging.

File: src/users/account-actions/PasswordHistory.test.jsx (hunk 1 above)
Comment (Ali-D-Akbar) [reply src/users/account-actions/PasswordHistory.test.jsx]: Ah, I missed those. Removing them.

**Review by azanbinzahid — APPROVED** [review]
Toggling password status refreshes whole user data, is it expected? If yes then it is good to go.

**Comment by Ali-D-Akbar** [comment]
> Toggling password status refreshes whole user data, is it expected? If yes then it is good to go.

Yes, it is the expected. We will revisit this probably in the near future.

**Pull request merged**